announcement/models/announcement.py

```diff
@@ -96,7 +96,7 @@ def _inverse_specific_user_ids(self):
         itself"""
         for announcement in self:
             for user in announcement.specific_user_ids.filtered(
-                lambda x: announcement
+                lambda x, announcement: announcement
```

Comment:
I think the second variable needs to be initialized

```suggestion
                lambda x, announcement=announcement: announcement
```

Reply:
done